### PR TITLE
Become a chef

### PR DESCRIPTION
**Pull request by giftofgrub**
This has all the commits from [https://github.com/hatchways/team-bluejay/pull/42](url)

To test:
1. run npm start
2. go to localhost:3000
3. open React devtools
4. login with dummy email and password
5. click on "become a chef"
6. supply a meal name, description (optional)
7. Check that the state in BecomeChefButton of the current user has changed to name => Chef, isChef => true
8. Check that the Become a Chef button is no longer there
9. Check the state of meals is no longer an empty array

### CODE REVIEW

File: client/src/components/BecomeChefButton.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      name: "mealName",
+      label: "Meal Name",
+      validation: {
+        required: "Name is required.",
+      },
+    },
+    {
+      name: "description",
+      label: "Description",
+    },
```
Comment (skassam21) [inline client/src/components/BecomeChefButton.js:46]: I think there is a bit more involved in the fields for a meal (such as the number of people this is for, the price of the meal).

**Review by skassam21 — COMMENTED** [review]
Looks good, but there is some overlapping changes with Kareem's PR, so I suggest we merge that one first and then re-review this PR

File: client/src/components/BecomeChefButton.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      
+      <Dialog
+          isOpen={dialogOpen}
+          handleClose={() => setDialogOpen(false)}
+      >
+        <Form
+          header="Add your first meal to become a chef"
+          onSubmit={handleSubmit}
+          fields={fields}
+          submitButtonLabel="Create Meal / Become A Chef"
```
Comment (kareemsakr) [inline client/src/components/BecomeChefButton.js]: I think Shums suggested changing the submit button to "Save" or "Submit"

File: client/src/contexts/AuthContext.js [hunk 3]
```diff
@@ -15,6 +15,8 @@ const reducer = (state, action) => {
       return { user: action.payload.user, errorMessage: "" };
     case "clearErrorMessage":
       return { ...state, errorMessage: "" };
+    case "refreshUser":
```
Comment (kareemsakr) [inline client/src/contexts/AuthContext.js:19]: You can remove this. Merge with the below branch and call refreshLoggedInUser on the auth context
https://github.com/hatchways/team-bluejay/tree/kareem-jwt-cookie-authentication-%2315-%2316

File: client/src/contexts/AuthContext.js [hunk 4]
```diff
@@ -24,6 +26,8 @@ const Context = React.createContext();
 
 //fake user until real login flow has been implemented
 const fakeUser = { name: "John Smith" };
+//fake chef until real flow has been implemented
+const fakeChef = { name: "Chef" , isChef: true};
```
Comment (kareemsakr) [inline client/src/contexts/AuthContext.js]: will not be needed once the above comment has been addressed

File: client/src/contexts/AuthContext.js [hunk 5]
```diff
@@ -55,9 +59,14 @@ const Provider = ({ children }) => {
     history.push("/login");
   };
 
+  const refreshUser = async() => {
```
Comment (kareemsakr) [inline client/src/contexts/AuthContext.js]: should be removed

File: client/src/contexts/MealContext.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    meals: [],
+    errorMessage: "",
+  });
+
+  let history = useHistory();
+  let location = useLocation();
+
+
+  const createMeal = async({mealName, description}) => {
+    // do backend call
```
Comment (kareemsakr) [inline client/src/contexts/MealContext.js:25]: the backend end point for creating a meal exists right ? is there a reason why you're not calling it ?

File: client/src/contexts/MealContext.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    dispatch({ 
+      type: "createMeal", 
+      payload: { createdMeal: 
+        { mealName, description }
+    }});
+    // go back to previous page
+    let { from } = location.state || { from: { pathname: "/" } };
```
Comment (kareemsakr) [inline client/src/contexts/MealContext.js]: why do we need to navigate back here ? we create a meal using a dialog right ? that does not affect the current route if am not mistaken

Comment (giftofgrub) [reply client/src/contexts/MealContext.js]: true, i will change this.

File: server/api/login_handler.py [hunk 8]
```diff
@@ -42,7 +42,9 @@ def post(self):
             refresh_token = create_refresh_token(
                 identity={"id": user.id}, expires_delta=timedelta(days=30))
             response = custom_json_response({
-                "message": "Authenticated", 'user': user_schema.dump(user)}, 200)
+                "message": "Authenticated", 
+                'user': user_schema.dump(user)}
+            , 200)
```
Comment (kareemsakr) [inline server/api/login_handler.py:47]: try to avoid committing these re-formats in the future

**Review by kareemsakr — APPROVED** [review]

**Pull request merged**